power_grid_model_c/power_grid_model_cpp/include/power_grid_model_cpp/buffer.hpp

```diff
@@ -0,0 +1,55 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#pragma once
+#ifndef POWER_GRID_MODEL_CPP_BUFFER_HPP
+#define POWER_GRID_MODEL_CPP_BUFFER_HPP
+
+#include "power_grid_model_c/buffer.h"
+
+#include "basics.hpp"
+#include "handle.hpp"
+
+namespace power_grid_model_cpp {
+class Buffer {
+  public:
+    Buffer(PGM_MetaComponent const* component, Idx size)
+        : component_{component}, size_{size}, buffer_{PGM_create_buffer(handle_.get(), component, size)} {};
+
+    void* get() const { return buffer_.get(); }
+
+    static void set_nan(Handle const& handle, PGM_MetaComponent const* component, Buffer& buffer, Idx buffer_offset,
```

Comment:
how would the user use this? maybe we need overloads after all..., even if only to add them to a custom namespace

Reply:
I will make `handle` `static` and remove it from the function signature, component can be obtained from `MetaData.get_component...` and the problem would then be how to get a `Buffer` for a free function, right? I would then probably have to add a `static` function to create and return a `Buffer`. I did not realize it before, but wouldn't that solve the usage issue? Why would we need overloads still?

Reply:
you can get `handle` directly from the member variables. since this function is declared in the class body, even if it's static, you have access to private member variables